cellbrowser_tools/fov_processing.py

```diff
@@ -1,28 +1,286 @@
 from aicsimageio.writers import OmeTiffWriter
 from aicsimageio.writers import PngWriter
-from aicsimageio.vendor.omexml import OMEXML
-from aicsimageio.vendor.omexml import qn
 from aicsimageio import AICSImage
 from . import cellJob
 from . import dataHandoffUtils as utils
 from .dataset_constants import DataField
 from aicsimageprocessing import thumbnailGenerator
 from aicsimageprocessing import textureAtlas
 
+# import copy
 import argparse
 import collections
-import copy
+from copy import deepcopy
 import errno
 import json
 import logging
 import numpy as np
+from ome_types import from_xml, to_xml
+from ome_types.model import Channel, TiffData, Plane
 import os
+import re
 import sys
+from tifffile import TiffFile
 import traceback
+import xml.etree.ElementTree as ET
+
 
 log = logging.getLogger()
 
 
+###############################################################################
+
+
+def check_num_planes(omepixels):
+    if len(omepixels.planes) != omepixels.size_c * omepixels.size_t * omepixels.size_z:
+        raise ValueError(
+            f"number of planes {len(omepixels.planes)} not consistent with sizeC*sizeZ*sizeT {omepixels.size_c}*{omepixels.size_t}*{omepixels.size_z}"
+        )
+
+
+def _clean_ome_xml_for_known_issues(xml: str) -> str:
```

Comment:
this is the function (yes it's huge) borrowed from future aicsimageio, and goes away when we transition to 4.0 api